cli/src/template.pest

```diff
@@ -25,7 +25,7 @@ raw_literal = @{ literal_char+ }
 literal = { "\"" ~ (raw_literal | escape)* ~ "\"" }
 
 integer_literal = {
-  ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*
+  "-"? ~ ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*
   | "0"
 }
```

Comment:
I would move this to it's own commit. That commit should also test that the places we currently accept integers fail in reasonable ways if you pass in a negative integer.  For example, before this change, we do this:

```
$ jj log -r @ -T 'change_id.short(-10)'
Error: Failed to parse template:  --> 1:17
  |
1 | change_id.short(-10)
  |                 ^---
  |
  = expected template
```

What's the new behavior? I suspect we'll want to update all the existing users of `expect_integer_expression()` to call some new `expect_positive_integer_expression()` helper.

Reply:
It will fall back to default or clamped at 0, I think. We'll need a runtime error to report this kind of errors properly.

Anyway, adding tests will help.

Reply:
Added tests, looks like the existing functions that take numerics all did what _I_ expected when passed a negative number.